integration-tests/tests/confirmDownloadPage.spec.ts

```diff
@@ -0,0 +1,106 @@
+import { AxiosResponse } from 'axios'
+import { sendRequest } from './utils/request/sendRequest'
+import { parse } from 'node-html-parser'
+import crypto from 'crypto'
+import { TriggerEndOfFlowSQSPayload } from './utils/types/sqsPayload'
+import { getIntegrationTestEnvironmentVariable } from './utils/getIntegrationTestEnvironmentVariable'
+import { invokeSQSOperationsLambda } from './utils/aws/invokeSQSOperationsLambdaFunction'
+import { pollNotifyMockForDownloadUrl } from './utils/notify/pollNotifyMockForDownloadUrl'
+
+describe('Confirm download page', () => {
+  let randomId = ''
+  let fileContents = ''
+
+  beforeEach(async () => {
+    randomId = crypto.randomUUID()
+    fileContents = crypto.randomUUID()
+
+    const payload: TriggerEndOfFlowSQSPayload = {
+      message: { athenaQueryId: randomId, fileContents: fileContents },
+      queueUrl: getIntegrationTestEnvironmentVariable(
+        'INTEGRATION_TESTS_TRIGGER_QUEUE_URL'
+      )
+    }
+    await invokeSQSOperationsLambda(payload)
+  })
+
+  it('A POST should return a success response when download url is valid', async () => {
+    const downloadUrl = await pollNotifyMockForDownloadUrl(randomId)
+
+    const response = await sendRequest(downloadUrl, 'POST')
+    expect(response.status).toEqual(200)
+
+    const fileDownloadResponse = await sendRequest(
+      retrieveS3LinkFromHtml(response.data),
+      'GET'
+    )
+    expect(fileDownloadResponse.status).toEqual(200)
+    expect(fileDownloadResponse.data as string).toEqual(fileContents)
+
+    const getResponseAfterDownload = await sendRequest(downloadUrl, 'GET')
+    expect(getResponseAfterDownload.status).toEqual(200)
+    expect(getResponseAfterDownload.data).toContain(
+      'You have 2 downloads remaining.'
+    )
+  })
+
+  it('A POST should return a 404 when there are no downloads remaining', async () => {
+    const downloadUrl = await pollNotifyMockForDownloadUrl(randomId)
+
+    const response = await sendRequest(downloadUrl, 'POST')
+    expect(response.status).toEqual(200)
+
+    let fileDownloadResponse = null
+    let getResponse = null
+    const s3Link = retrieveS3LinkFromHtml(response.data)
+    fileDownloadResponse = await sendRequest(s3Link, 'GET')
+    expect(fileDownloadResponse.status).toEqual(200)
+    expect(fileDownloadResponse.data as string).toEqual(fileContents)
+
+    for (let i = 2; i >= 0; i--) {
```

Comment:
I think you could make the tests run quite a lot faster if you simply set up a download at the start of all the tests, then do the assertions for your first test.

Instead what you're doing is creating a record, then doing the whole thing again when you want to make sure the download count goes down.

Reply:
Done.